modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -275,7 +310,9 @@ public void start() {
 
         logicalTopologyService.addEventListener(topologyEventListener);
 
-        initMetaStorageKeysOnStart();
+        registerMetaStorageWatchListener()
```

Comment:
What will happen in case of concurrent start/stop calls?
Busy locks are missing almost everywhere.

Reply:
I added busy lock.